libsolidity/formal/BMC.cpp

```diff
@@ -591,6 +591,7 @@ void BMC::internalOrExternalFunctionCall(FunctionCall const& _funCall)
 	{
 		m_externalFunctionCallHappened = true;
 		resetStorageVariables();
+		resetBalances();
```

Comment:
This is the balance of the current contract, isn't it? Do we have to erase it for all calls other than `internal`? If the value field is not set for regular `call`, we don't need this. Similarly, for `staticcall`.

Reply:
I see that maybe a callback can change balances for a regualar `call`. But still for `staticall`, we shouldn't change this.

Reply:
Why would we erase it for `internal` calls? Those get inlined at the beginning of the function.
But yea, we could filter it better.
I would do the value field in a different PR, targeting https://github.com/ethereum/solidity/issues/11249
Maybe it makes sense to do both that and `staticcall` in the same PR?

I kind of want to keep this one here minimal, even if it introduces false positives (but still sound), to move on with the other PR which is more important.

Reply:
I only saw your second comment after writing the above.
I'd personally prefer doing the `staticcall` part in a different PR, but don't have too strong of an opinion here. If you think it should be in this one I'll change it

Reply:
I meant other than `internal` in the sense that that was the condition for this else branch.

> Maybe it makes sense to do both that and staticcall in the same PR

Yeah, isn't that change simple though? adding another `else if (funType.kind() == FunctionType::Kind::Staticcall) { // dont' reset value or state }`

Reply:
It's simple, but I'm annoyed by the tests duration and blocking the other PR for another 2h (at least) which is in turn blocking another PR. But I'll change it.

Reply:
If CI is the problem, then I don't mind merging this and fixing staticall later.

Reply:
It's fine, I'll just do some other stuff in the meantime. Running the tests locally, will probably add some more to enforce that BMC is running on staticcall